crates/wash-lib/src/cli/start.rs

```diff
@@ -72,9 +73,111 @@ pub struct StartComponentCommand {
     /// If this flag is omitted, the timeout will be adjusted to 5 seconds to account for component download times
     #[clap(long = "skip-wait")]
     pub skip_wait: bool,
+
+    /// Digest to verify artifact against (if OCI URL is provided for <target>)
+    #[clap(short = 'd', long = "digest")]
+    pub digest: Option<String>,
+
+    /// Allow latest artifact tags (if OCI URL is provided for <target>)
+    #[clap(long = "allow-latest")]
+    pub allow_latest: bool,
+
+    /// OCI username, if omitted anonymous authentication will be used
+    #[clap(long = "user", env = "WASH_REG_USER", hide_env_values = true)]
+    pub user: Option<String>,
+
+    /// OCI password, if omitted anonymous authentication will be used
+    #[clap(long = "password", env = "WASH_REG_PASSWORD", hide_env_values = true)]
+    pub password: Option<String>,
+
+    /// Allow insecure (HTTP) registry connections
+    #[clap(long = "insecure")]
+    pub insecure: bool,
+
+    /// skip the local OCI cache and pull the artifact from the registry to inspect
+    #[clap(long = "no-cache")]
+    pub no_cache: bool,
 }
 
-pub async fn handle_start_component(cmd: StartComponentCommand) -> Result<CommandOutput> {
+pub async fn handle_start_component(cmd: StartCommand) -> Result<CommandOutput> {
+    // TODO: absolutize the path if it's a relative file
+    let component_ref = if cmd.component_ref.starts_with('/') {
+        format!("file://{}", &cmd.component_ref) // prefix with file:// if it's an absolute path
+    } else {
+        cmd.component_ref.to_string()
+    };
+
+    let mut buf = Vec::new();
+    if PathBuf::from(component_ref.clone()).as_path().is_dir() {
+        let mut f = File::open(component_ref.clone())?;
+        f.read_to_end(&mut buf)?;
+    } else {
+        let cache_file = (!cmd.no_cache).then(|| cached_oci_file(&cmd.component_ref.clone()));
+        buf = get_oci_artifact(
+            component_ref.clone(),
+            cache_file,
+            OciPullOptions {
+                digest: cmd.digest.clone(),
+                allow_latest: cmd.allow_latest,
+                user: cmd.user.clone(),
+                password: cmd.password.clone(),
+                insecure: cmd.insecure,
+            },
+        )
+        .await?;
+    }
+
+    let provider = ProviderArchive::try_load(&buf).await;
+
+    if provider.is_ok() {
+        return handle_start_provider(cmd)
+            .await
+            .with_context(|| format!("Failed to start provider component: {}", component_ref));
+    }
+
+    let actor = match wasmparser::Parser::new(0).parse_all(&buf).next() {
+        // Inspect claims inside of Wasm
+        Some(Ok(wasmparser::Payload::Version {
```

Comment:
We don't need to do any of this claims inspection as it will be done by the host (plus claims are optional now). I would just check if parses like you do here and then continue on